build.gradle

```diff
@@ -234,76 +235,99 @@ java {
-    client {
-    }
-    clientAlt {
-        configure('client')
-        //Force disable devLogin for clientAlt regardless of if it is enabled via gradle properties for the main gradle run
-        devLogin.enabled(false)
-        if (!(findProperty('neogradle.subsystems.devLogin.conventionForRun') ?: false)) {
-            //If the property is missing or set to false (so the normal runClient task would use Dev as the name),
-            // change the name of the alt client type so that it doesn't conflict with the main dev one
-            programArguments.addAll((String[]) ['--username', 'AltDev'])
-        }
-    }
-    server {
-    }
-    gameTestServer {
-        modSources.add((SourceSet) sourceSets.gameTest)
-    }
-    gameTestClient {
-        configure('client')
-        modSources.add((SourceSet) sourceSets.gameTest)
-    }
-    junit {
-        unitTestSources.add((SourceSet) sourceSets.test)
+    for (String name : secondaryModules) {
+        mods.maybeCreate(name).modSourceSets.add(sourceSets.named(name))
     }
-    data {
-        programArguments.addAll((String[]) ['--all', '--output', file('src/datagen/generated/').absolutePath,
-                                            '--mod', 'mekanism', '--existing', file('src/main/resources/').absolutePath])
 
-        modSources.add((SourceSet) sourceSets.datagenMain)
+    accessTransformers = [
+            'src/main/resources/META-INF/accesstransformer.cfg',
+            'src/additions/resources/META-INF/accesstransformer.cfg',
+            //Dev time only ATs so the file name doesn't have to match accesstransformer.cfg
+            'src/datagen/main/resources/META-INF/datagen_ats.cfg',
+            'src/gameTest/resources/META-INF/gametest_ats.cfg',
+    ]
 
-        for (String name : secondaryModules) {
-            modSources.add(getExtraSourceSet(name, 'datagen').get())
-            programArguments.addAll((String[]) ['--mod', "mekanism${name}", '--existing', file("src/${name}/resources/").absolutePath])
-        }
+    runs {
+        configureEach {
+            if (hasProperty('forge_force_ansi')) {
+                //Force ansi if declared as a gradle variable, as the auto detection doesn't detect IntelliJ properly
+                // or eclipse's plugin that adds support for ansi escape in console
+                systemProperties.put('terminal.ansi', (String) property('forge_force_ansi'))
+            }
+
+            mods.add(neoForge.mods.named("mekanism"))
+            for (String name : secondaryModules) {
+                mods.add(neoForge.mods.named(name))
+            }
 
-        dependencies {
-            runtime(configurations.datagenNonMod)
+            //if the selected toolchain is a JBR, enable DCEVM
+            if (javaToolchains.launcherFor(java.toolchain).map { it.metadata.vendor }.getOrElse('').contains('JetBrains')) {
+                // TODO MDG jvmArguments.add('-XX:+AllowEnhancedClassRedefinition')
```

Comment:
🤔

Reply:
this is possible now.
you just do `jvmArgument('-XX:+AllowEnhancedClassRedefinition')`

Reply:
uhh, i thought jvms were supposed to ignore unknown XX params :/

Reply:
I think that is only the case with `-XX:+IgnoreUnrecognizedVMOptions`